merkle/src/merkle.rs

```diff
@@ -198,8 +216,24 @@ impl<T: Ord + Clone + AsRef<[u8]> + Send + Sync, A: Algorithm<T>> FromParallelIt
         let size = 2 * pow - 1;
 
         assert!(leafs > 1);
+
+//        let mmap_slice = &(*anonymous_mmap(size * mem::size_of::<T>()));
+//        unsafe {
+//            let parameter_slice = mem::transmute::<&[u8], &[T]>(mmap_slice);
+//            let adjusted_slice = slice::from_raw_parts(parameter_slice.as_ptr(), size);
+//            adjusted_slice.to_vec()
+//        }
+        // FIXME: pass contents of data into the `mmap` box/slice.
```

Comment:
Need to figure out how to replace one `Box` (from the `Vec`) with another (from `mmap`). The main difficulty at the moment is that `data` is `Box<[T]>`  but `mmap` only works with references like `Box<&[T]>` (otherwise it complains about not knowing the size at compile time). Changing `data` to `Box<&[T]>` forces to also change it to `Box<&mut [T]>` (to perform the same `push`es as with `Vec`) and that in turn brings other complications. (Also, it's not really `Box<&mut [T]>` but rather `Box<&'a mut [T]>`, that is, I have to start adding lifetimes all over the implementation.)